src/handlers/chat-input-command-handler.js

```diff
@@ -10,14 +11,25 @@ async function chatInputCommandHandler(interaction) {
 	/** @type {Discord.Collection} */
 	const commands = interaction.client.commands;
 	const command = commands.get(commandName);
+	const memberId = interaction.member.id
 
 	// do nothing if no command
 	if (!command) {
 		throw new Error(`Missing command handler for \`${commandName}\``);
 	}
 
-	// run the command
-	await command.handler(interaction);
+	// check for cooldown and run the command
+	const cooldown = await cooldownUtils.isInteractionOnCooldown(command, memberId)
+	if(!cooldown) 
+	{
+		await command.handler(interaction);
+		if(command.cooldown) { cooldownUtils.beginCooldown(command, memberId) }
+	} else {
+	await interaction.reply({
+		embeds: [cooldown],
+		ephemeral: true
+	});
+	}
```

Comment:
Same formatting comments as before. `if` needs a space, brackets should be on the same line (`if (true) {`), `if` statements are _always_ multi-line, and the interaction reply has bad indentation